flux-middle/src/rustc/lowering.rs

```diff
@@ -558,6 +558,20 @@ pub fn lower_variant_def(
 pub fn lower_fn_sig_of(tcx: TyCtxt, def_id: DefId) -> Result<PolyFnSig, errors::UnsupportedFnSig> {
     let fn_sig = tcx.fn_sig(def_id);
     let span = tcx.def_span(def_id);
+    let fn_sig = if !fn_sig.has_projections() {
+        fn_sig
+    } else {
+        let param_env = tcx.param_env_reveal_all_normalized(def_id);
+        match tcx.try_normalize_erasing_regions(param_env, fn_sig) {
+            Ok(fn_sig) => fn_sig,
+            Err(_) => {
+                return Err(errors::UnsupportedFnSig {
+                    span,
+                    reason: "Sorry, projections are not yet supported!".to_string(),
+                })
+            }
+        }
+    };
```

Comment:
We are not doing anything interesting with regions yet, but we want to, so we shouldn't be erasing them. For example, the following fails right now

```rust
trait Trait {
    type A;
}

impl Trait for i32 {
    type A = i32;
}

fn foo(x: &i32) -> <i32 as Trait>::A {
    *x
}
```

I don't understand exactly what's going on with the obligations and InferCtxt but the following seems to work without erasing the regions

```rust
pub fn lower_fn_sig_of(tcx: TyCtxt, def_id: DefId) -> Result<PolyFnSig, errors::UnsupportedFnSig> {
    let fn_sig = tcx.fn_sig(def_id);
    let span = tcx.def_span(def_id);
    let param_env = tcx.param_env(def_id);
    match tcx
        .infer_ctxt()
        .build()
        .at(&ObligationCause::dummy(), param_env)
        .normalize(fn_sig)
    {
        Ok(fn_sig) => {
            lower_fn_sig(tcx, fn_sig.value)
                .map_err(|err| errors::UnsupportedFnSig { span, reason: err.reason })
        }
        Err(_) => {
            return Err(errors::UnsupportedFnSig {
                span,
                reason: "Sorry, projections are not yet supported!".to_string(),
            })
        }
    }
}
```